ralph/src/main/scala/org/alephium/ralph/Compiler.scala

```diff
@@ -247,7 +247,10 @@ object Compiler {
       if (inputType == argsType) {
         returnType
       } else {
-        throw Error(s"Invalid args type $inputType for function $name", id.sourceIndex)
+        throw Error(
+          s"Invalid args type ${quote(inputType)} for func $name, expected ${quote(argsType)}",
+          None
```

Comment:
you removed the `id.sourceIndex` is it on purpose? I'm just not sure this error will be catch to get the position added

Reply:
Yeah, since this is called during the function call and we want the error position to be set at the call site instead of the function itself.